File: backend/internal/models/product.go
package models

type Product struct {
	ID           uint                 `gorm:"primaryKey" json:"id"`
	CategoryID   uint                 `gorm:"not null" json:"category_id"`
	Category     Category             `json:"category,omitempty"`
	Name         string               `gorm:"size:100;not null" json:"name"`
	Slug         string               `gorm:"size:100;unique;not null" json:"slug"`
	Description  string               `gorm:"type:text" json:"description"`
	Price        float64              `gorm:"not null" json:"price"`
	ImageURL     string               `gorm:"size:255" json:"image_url"`
	IsAvailable  bool                 `gorm:"default:true" json:"is_available"`
	OptionGroups []ProductOptionGroup `json:"option_groups,omitempty"`
}

File: mobile/src/types/product.ts
export interface Category {
  id: number;
  name: string;
  slug: string;
}

export interface Product {
  id: number;
  name: string;
  price: number;
  image_url: string;
  is_available: boolean;
}

export interface ProductDetail extends Product {
  description: string;
  category_name: string;
  option_groups: OptionGroup[];
}

export interface OptionGroup {
  id: number;
  name: string;
  selection_type: string;
  is_required: boolean;
  values: OptionValue[];
}

export interface OptionValue {
  id: number;
  label: string;
  extra_price: number;
}
